### PR TITLE
fix: The feature test engine now throws error when schema mismatch

### PR DESCRIPTION
**Pull request by energinet-ajw**
Before this fix the feature test engine failed to throw errors if there was a schema naming mismatch between the actual schema and the loaded CSV file's column names.

## Pull-request quality

- [x] The title adheres to [this guide](https://github.com/Mech0z/GitHubGuidelines)
- [ ] Tests are written and executed locally
- [ ] Subsystem tests have been tested (by manually deploying to `dev_002`)

### CODE REVIEW

File: source/databricks/calculation_engine/tests/features/utils/scenario_executor.py [hunk 1]
```diff
@@ -68,19 +69,32 @@ def _setup(self, scenario_path: str) -> None:
...
-        # We need to create the dataframe again because nullability and precision
-        # are not applied when reading the csv file.
-        return spark_session.createDataFrame(df.rdd, schema)
+        # Verify column names match with those in the schema
+        schema_column_names = [field.name for field in schema.fields]
+        if schema_column_names != df.columns:
+            raise ValueError(
+                f"Column names in {path_to_csv} do not match with the schema. "
+                f"Expected: {schema_column_names}, Found: {df.columns}"
+            )
```
Comment (BjarkeMeier) [inline source/databricks/calculation_engine/tests/features/utils/scenario_executor.py:89]: Don't we already have a function that implements this functionality?

File: source/databricks/calculation_engine/tests/features/utils/scenario_executor.py [hunk 2]
```diff
@@ -68,19 +69,32 @@ def _setup(self, scenario_path: str) -> None:
...
+
+        # Read the CSV file again, this time enforcing the schema
+        df = spark_session.read.csv(
+            path_to_csv, header=True, sep=";", schema=schema, enforceSchema=True
+        )
```
Comment (BjarkeMeier) [inline source/databricks/calculation_engine/tests/features/utils/scenario_executor.py:94]: 1. What do you get by enforcing the schema?
2. Does it work for CSV files? I think I once learned that it doesn't.

Comment (energinet-ajw) [reply source/databricks/calculation_engine/tests/features/utils/scenario_executor.py:94]: I think you are right. I will remove the enforcement and update the comment.

File: source/databricks/calculation_engine/tests/features/utils/scenario_executor.py (hunk 1 above)
Comment (energinet-ajw) [reply source/databricks/calculation_engine/tests/features/utils/scenario_executor.py:89]: The assert_schema function has logic that tests column names, but I can't use it directly.

**Review by johevemi — APPROVED** [review]

**Pull request merged**